atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java

```diff
@@ -132,7 +125,7 @@ public TransactionScopedCache createTransactionScopedCache(long startTs) {
      *     ever have new events, and that consecutive calls to this method will *always* have increasing sequences
      *     (without this last guarantee, we'd need to store snapshots for all sequences).
      */
-    private void updateStores(TransactionsLockWatchUpdate updateForTransactions) {
+    private synchronized void updateStores(TransactionsLockWatchUpdate updateForTransactions) {
```

Comment:
Synchronised to be safe - these should be executed within synchronised methods anyway, but since not every public method on this class is synchronised, I wanted to be doubly certain.